### PR TITLE
[New Rule] Global Administrator Role Assigned

### LINKED ISSUE #1685: [New Rule] Global Administrator Role Assigned
## Description

In Azure Active Directory (Azure AD), permissions to manage resources are assigned using Roles. The Global Administrator
is a role that enables users to have access to all administrative features in Azure AD and services that use Azure
Active Directory identities like the Microsoft 365 Defender portal, the Microsoft 365 compliance center, Exchange,
SharePoint Online, and Skype for Business Online. Attackers can add users as Global Administrators to maintain access
and manage all subscriptions and their settings and resources.

### Target indexes
"filebeat-*", "logs-azure*"

### Query

Microsoft 365/Office 365 audit:
```
event.dataset:o365.audit and event.code:"AzureActiveDirectory" and event.action: "Add member to role." and
o365.audit.ModifiedProperties.Role_DisplayName.NewValue: "Global Administrator"
```

Azure audit logs:
```
event.dataset:azure.auditlogs and azure.auditlogs.properties.category:RoleManagement and
azure.auditlogs.operation_name : "Add member to role" and
azure.auditlogs.properties.target_resources.0.modified_properties.1.new_value: "\"Global Administrator\""
```

## Example Data
Azure AD:
```
{
  "_index": ".ds-logs-azure.auditlogs-default-2022.01.04-000001",
  "_type": "_doc",
  "_id": "JSMIMH4B7Kld_xqUWl6l",
  "_version": 1,
  "_score": 1,
  "_source": {
    "agent": {
      "hostname": "Shipper",
      "name": "Shipper",
      "id": "21d0ccfd-4784-4126-8445-cc4c3437e6e4",
      "ephemeral_id": "684b66ad-a186-417e-b4a5-eb37263289d1",
      "type": "filebeat",
      "version": "7.16.2"
    },
    "elastic_agent": {
      "id": "21d0ccfd-4784-4126-8445-cc4c3437e6e4",
      "version": "7.16.2",
      "snapshot": false
    },
    "source": {
      "geo": {
        "continent_name": "South America",
        "region_iso_code": "BR-SP",
        "city_name": "Campinas",
        "country_iso_code": "BR",
        "country_name": "Brazil",
        "region_name": "Sao Paulo",
        "location": {
          "lon": -47.0565,
          "lat": -22.9035
        }
      },
      "as": {
        "number": 8075,
        "organization": {
          "name": "MICROSOFT-CORP-MSN-AS-BLOCK"
        }
      },
      "ip": "42.126.45.24"
    },
    "azure-eventhub": {
      "sequence_number": 31,
      "consumer_group": "$Default",
      "offset": 113872,
      "eventhub": "hubb",
      "enqueued_time": "2022-01-06T15:35:59.859Z"
    },
    "tags": [
      "azure-auditlogs"
    ],
    "cloud": {
      "instance": {
        "name": "Shipper",
        "id": "3dc43ef6-ee8b-4e06-895b-d134cb4513c8"
      },
      "provider": "azure",
      "machine": {
        "type": "Standard_D2s_v3"
      },
      "service": {
        "name": "Virtual Machines"
      },
      "region": "eastus",
      "account": {}
    },
    "input": {
      "type": "azure-eventhub"
    },
    "@timestamp": "2022-01-06T15:33:40.165Z",
    "ecs": {
      "version": "1.12.0"
    },
    "related": {
      "ip": [
        "42.126.45.24"
      ]
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "azure.auditlogs"
    },
    "host": {
      "hostname": "Shipper",
      "os": {
        "kernel": "5.11.0-1023-azure",
        "codename": "focal",
        "name": "Ubuntu",
        "family": "debian",
        "type": "linux",
        "version": "20.04.3 LTS (Focal Fossa)",
        "platform": "ubuntu"
      },
      "containerized": false,
      "ip": [
        "10.0.0.4",
        "fe80::20d:3aff:fe8d:e33a"
      ],
      "name": "Shipper",
      "id": "1fda7c18d5b14fccb2a9772187abbe03",
      "mac": [
        "00:0d:3a:8d:e3:3a",
        "00:0d:3a:8d:e3:3a"
      ],
      "architecture": "x86_64"
    },
    "client": {
      "ip": "42.126.45.24"
    },
    "event": {
      "duration": 0,
      "agent_id_status": "verified",
      "ingested": "2022-01-06T15:36:00Z",
      "kind": "event",
      "action": "Add member to role",
      "dataset": "azure.auditlogs",
      "outcome": "success"
    },
    "azure": {
      "tenant_id": "1c04d22a-c0e8-4bec-840b-f1a7fb855984",
      "auditlogs": {
        "operation_name": "Add member to role",
        "operation_version": "1.0",
        "Level": 4,
        "category": "AuditLogs",
        "result_signature": "None",
        "properties": {
          "logged_by_service": "Core Directory",
          "initiated_by": {
            "user": {
              "displayName": null,
              "roles": [],
              "ipAddress": "42.126.45.24",
              "id": "892daa3c-4f3a-4a1a-b4ef-0fc987d61634",
              "userPrincipalName": "Test1@domain.com"
            }
          },
          "activity_display_name": "Add member to role",
          "operation_type": "Assign",
          "additional_details": [],
          "correlation_id": "b566a715-8e02-4c1d-9b69-86c18da3d192",
          "userAgent": null,
          "activity_datetime": "2022-01-06T15:33:40.1650779+00:00",
          "target_resources": {
            "0": {
              "user_principal_name": "testuser@domain.com",
              "modified_properties": {
                "0": {
                  "old_value": null,
                  "display_name": "Role.ObjectID",
                  "new_value": "\"a6ce8341-0ca1-4f04-8127-7d891f95103c\""
                },
                "1": {
                  "old_value": null,
                  "display_name": "Role.DisplayName",
                  "new_value": "\"Global Administrator\""
                },
                "2": {
                  "old_value": null,
                  "display_name": "Role.TemplateId",
                  "new_value": "\"62e90394-69f5-4237-9190-012177145e10\""
                },
                "3": {
                  "old_value": null,
                  "display_name": "Role.WellKnownObjectName",
                  "new_value": "\"TenantAdmins\""
                }
              },
              "id": "255d2df7-4f0f-40e9-a14d-445c80b2cfbf",
              "type": "User"
            },
            "1": {
              "modified_properties": {},
              "id": "a6ce8341-0ca1-4f04-8127-7d891f95103c",
              "type": "Role"
            }
          },
          "id": "Directory_b566a715-8e02-4c1d-9b69-86c18da3d192_XR6Q1_42847322",
          "category": "RoleManagement",
          "result_reason": ""
        }
      },
      "resource": {
        "provider": "Microsoft.aadiam",
        "id": "/tenants/1c04d22a-c0e8-4bec-840b-f1a7fb855984/providers/Microsoft.aadiam"
      },
      "correlation_id": "b566a715-8e02-4c1d-9b69-86c18da3d192"
    }
  },
  "fields": {
    "azure.auditlogs.properties.target_resources.0.type": [
      "User"
    ],
    "azure.auditlogs.operation_version": [
      "1.0"
    ],
    "azure.resource.provider": [
      "Microsoft.aadiam"
    ],
    "azure.resource.id": [
      "/tenants/1c04d22a-c0e8-4bec-840b-f1a7fb855984/providers/Microsoft.aadiam"
    ],
    "elastic_agent.version": [
      "7.16.2"
    ],
    "azure.auditlogs.properties.initiated_by.user.ipAddress": [
      "42.126.45.24"
    ],
    "host.hostname": [
      "Shipper"
    ],
    "host.mac": [
      "00:0d:3a:8d:e3:3a",
      "00:0d:3a:8d:e3:3a"
    ],
    "azure.auditlogs.properties.correlation_id": [
      "b566a715-8e02-4c1d-9b69-86c18da3d192"
    ],
    "host.os.version": [
      "20.04.3 LTS (Focal Fossa)"
    ],
    "source.geo.region_name": [
      "Sao Paulo"
    ],
    "host.os.name": [
      "Ubuntu"
    ],
    "source.ip": [
      "42.126.45.24"
    ],
    "agent.name": [
      "Shipper"
    ],
    "host.name": [
      "Shipper"
    ],
    "event.agent_id_status": [
      "verified"
    ],
    "source.geo.region_iso_code": [
      "BR-SP"
    ],
    "event.kind": [
      "event"
    ],
    "source.geo.city_name": [
      "Campinas"
    ],
    "event.outcome": [
      "success"
    ],
    "host.os.type": [
      "linux"
    ],
    "cloud.region": [
      "eastus"
    ],
    "azure.auditlogs.properties.initiated_by.user.userPrincipalName": [
      "Test1@domain.com"
    ],
    "azure.auditlogs.properties.activity_datetime": [
      "2022-01-06T15:33:40.165Z"
    ],
    "azure.auditlogs.properties.target_resources.1.id": [
      "a6ce8341-0ca1-4f04-8127-7d891f95103c"
    ],
    "input.type": [
      "azure-eventhub"
    ],
    "client.ip": [
      "42.126.45.24"
    ],
    "agent.hostname": [
      "Shipper"
    ],
    "azure.auditlogs.properties.target_resources.0.modified_properties.1.new_value": [
      "\"Global Administrator\""
    ],
    "data_stream.type": [
      "logs"
    ],
    "tags": [
      "azure-auditlogs"
    ],
    "azure.auditlogs.properties.id": [
      "Directory_b566a715-8e02-4c1d-9b69-86c18da3d192_XR6Q1_42847322"
    ],
    "host.architecture": [
      "x86_64"
    ],
    "cloud.machine.type": [
      "Standard_D2s_v3"
    ],
    "cloud.provider": [
      "azure"
    ],
    "azure.auditlogs.properties.operation_type": [
      "Assign"
    ],
    "agent.id": [
      "21d0ccfd-4784-4126-8445-cc4c3437e6e4"
    ],
    "cloud.service.name": [
      "Virtual Machines"
    ],
    "host.containerized": [
      false
    ],
    "ecs.version": [
      "1.12.0"
    ],
    "agent.version": [
      "7.16.2"
    ],
    "host.os.family": [
      "debian"
    ],
    "azure.auditlogs.properties.target_resources.1.type": [
      "Role"
    ],
    "source.as.number": [
      8075
    ],
    "azure.auditlogs.properties.activity_display_name": [
      "Add member to role"
    ],
    "azure.auditlogs.properties.target_resources.0.id": [
      "255d2df7-4f0f-40e9-a14d-445c80b2cfbf"
    ],
    "azure.auditlogs.properties.target_resources.0.modified_properties.0.new_value": [
      "\"a6ce8341-0ca1-4f04-8127-7d891f95103c\""
    ],
    "source.geo.location": [
      {
        "coordinates": [
          -47.0565,
          -22.9035
        ],
        "type": "Point"
      }
    ],
    "azure.auditlogs.category": [
      "AuditLogs"
    ],
    "host.ip": [
      "10.0.0.4",
      "fe80::20d:3aff:fe8d:e33a"
    ],
    "cloud.instance.id": [
      "3dc43ef6-ee8b-4e06-895b-d134cb4513c8"
    ],
    "agent.type": [
      "filebeat"
    ],
    "event.module": [
      "azure"
    ],
    "azure.auditlogs.properties.initiated_by.user.id": [
      "892daa3c-4f3a-4a1a-b4ef-0fc987d61634"
    ],
    "related.ip": [
      "42.126.45.24"
    ],
    "host.os.kernel": [
      "5.11.0-1023-azure"
    ],
    "source.geo.country_iso_code": [
      "BR"
    ],
    "azure-eventhub.eventhub": [
      "hubb"
    ],
    "azure.auditlogs.result_signature": [
      "None"
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "host.id": [
      "1fda7c18d5b14fccb2a9772187abbe03"
    ],
    "azure-eventhub.consumer_group": [
      "$Default"
    ],
    "azure.auditlogs.properties.target_resources.0.modified_properties.3.new_value": [
      "\"TenantAdmins\""
    ],
    "azure.auditlogs.properties.logged_by_service": [
      "Core Directory"
    ],
    "azure.auditlogs.properties.category": [
      "RoleManagement"
    ],
    "azure.tenant_id": [
      "1c04d22a-c0e8-4bec-840b-f1a7fb855984"
    ],
    "azure.auditlogs.properties.target_resources.0.user_principal_name": [
      "testuser@domain.com"
    ],
    "azure-eventhub.offset": [
      113872
    ],
    "elastic_agent.id": [
      "21d0ccfd-4784-4126-8445-cc4c3437e6e4"
    ],
    "azure.correlation_id": [
      "b566a715-8e02-4c1d-9b69-86c18da3d192"
    ],
    "azure.auditlogs.operation_name": [
      "Add member to role"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "host.os.codename": [
      "focal"
    ],
    "source.geo.continent_name": [
      "South America"
    ],
    "source.as.organization.name": [
      "MICROSOFT-CORP-MSN-AS-BLOCK"
    ],
    "azure-eventhub.enqueued_time": [
      "2022-01-06T15:35:59.859Z"
    ],
    "azure.auditlogs.properties.result_reason": [
      ""
    ],
    "azure.auditlogs.properties.target_resources.0.modified_properties.2.display_name": [
      "Role.TemplateId"
    ],
    "event.duration": [
      0
    ],
    "azure.auditlogs.properties.target_resources.0.modified_properties.0.display_name": [
      "Role.ObjectID"
    ],
    "event.ingested": [
      "2022-01-06T15:36:00.000Z"
    ],
    "event.action": [
      "Add member to role"
    ],
    "azure-eventhub.sequence_number": [
      31
    ],
    "@timestamp": [
      "2022-01-06T15:33:40.165Z"
    ],
    "host.os.platform": [
      "ubuntu"
    ],
    "azure.auditlogs.properties.target_resources.0.modified_properties.3.display_name": [
      "Role.WellKnownObjectName"
    ],
    "azure.auditlogs.properties.target_resources.0.modified_properties.2.new_value": [
      "\"62e90394-69f5-4237-9190-012177145e10\""
    ],
    "data_stream.dataset": [
      "azure.auditlogs"
    ],
    "agent.ephemeral_id": [
      "684b66ad-a186-417e-b4a5-eb37263289d1"
    ],
    "source.geo.country_name": [
      "Brazil"
    ],
    "azure.auditlogs.Level": [
      4
    ],
    "event.dataset": [
      "azure.auditlogs"
    ],
    "cloud.instance.name": [
      "Shipper"
    ],
    "azure.auditlogs.properties.target_resources.0.modified_properties.1.display_name": [
      "Role.DisplayName"
    ]
  }
}
```

Office 365:

```
{
  "_index": ".ds-logs-o365.audit-default-2022.01.04-000001",
  "_type": "_doc",
  "_id": "3f56f3e7-7098-45f2-b0fc-3cbad632120c",
  "_version": 1,
  "_score": 1,
  "_ignored": [
    "event.original"
  ],
  "_source": {
    "agent": {
      "hostname": "Shipper",
      "name": "Shipper",
      "id": "21d0ccfd-4784-4126-8445-cc4c3437e6e4",
      "type": "filebeat",
      "ephemeral_id": "684b66ad-a186-417e-b4a5-eb37263289d1",
      "version": "7.16.2"
    },
    "elastic_agent": {
      "id": "21d0ccfd-4784-4126-8445-cc4c3437e6e4",
      "version": "7.16.2",
      "snapshot": false
    },
    "tags": [
      "forwarded",
      "o365-audit",
      "preserve_original_event"
    ],
    "cloud": {
      "instance": {
        "name": "Shipper",
        "id": "3dc43ef6-ee8b-4e06-895b-d134cb4513c8"
      },
      "provider": "azure",
      "machine": {
        "type": "Standard_D2s_v3"
      },
      "service": {
        "name": "Virtual Machines"
      },
      "region": "eastus",
      "account": {}
    },
    "o365": {
      "audit": {
        "AzureActiveDirectoryEventType": "1",
        "ResultStatus": "Success",
        "ObjectId": "testuser@domain.com",
        "UserKey": "10032001C8CDE20F@domain.com",
        "ExtendedProperties": {
          "extendedAuditEventCategory": "Role",
          "additionalDetails": "{}"
        },
        "IntraSystemId": "182829f7-59da-47fd-95e2-466566f6ad47",
        "Target": [
          {
            "Type": 2,
            "ID": "User_255d2df7-4f0f-40e9-a14d-445c80b2cfbf"
          },
          {
            "Type": 2,
            "ID": "255d2df7-4f0f-40e9-a14d-445c80b2cfbf"
          },
          {
            "Type": 2,
            "ID": "User"
          },
          {
            "Type": 5,
            "ID": "testuser@domain.com"
          },
          {
            "Type": 3,
            "ID": "10032001C98098E7"
          }
        ],
        "RecordType": "8",
        "Version": "1",
        "ModifiedProperties": {
          "Role_DisplayName": {
            "OldValue": "",
            "NewValue": "Global Administrator"
          },
          "Role_TemplateId": {
            "OldValue": "",
            "NewValue": "62e90394-69f5-4237-9190-012177145e10"
          },
          "Role_WellKnownObjectName": {
            "OldValue": "",
            "NewValue": "TenantAdmins"
          },
          "Role_ObjectID": {
            "OldValue": "",
            "NewValue": "a6ce8341-0ca1-4f04-8127-7d891f95103c"
          }
        },
        "SupportTicketId": "",
        "UserId": "User1@domain.com",
        "TargetContextId": "1c04d22a-c0e8-4bec-840b-f1a7fb855091",
        "Actor": [
          {
            "Type": 5,
            "ID": "User1@domain.com"
          },
          {
            "Type": 3,
            "ID": "10032001C8CDE20F"
          },
          {
            "Type": 2,
            "ID": "User_892daa3c-4f3a-4a1a-b4ef-0fc987d61634"
          },
          {
            "Type": 2,
            "ID": "892daa3c-4f3a-4a1a-b4ef-0fc987d61634"
          },
          {
            "Type": 2,
            "ID": "User"
          }
        ],
        "CreationTime": "2022-01-06T15:33:40",
        "InterSystemsId": "b566a715-8e02-4c1d-9b69-86c18da3d194",
        "UserType": "0",
        "ActorContextId": "1c04d22a-c0e8-4bec-840b-f1a7fb855091"
      }
    },
    "input": {
      "type": "o365audit"
    },
    "@timestamp": "2022-01-06T15:33:40.000Z",
    "ecs": {
      "version": "1.12.0"
    },
    "related": {
      "user": [
        "User1",
        "testuser"
      ]
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "o365.audit"
    },
    "organization": {
      "id": "1c04d22a-c0e8-4bec-840b-f1a7fb855091"
    },
    "host": {
      "name": "domain.com",
      "id": "1c04d22a-c0e8-4bec-840b-f1a7fb855091"
    },
    "event": {
      "agent_id_status": "verified",
      "ingested": "2022-01-06T15:59:15Z",
      "original": "{\"CreationTime\":\"2022-01-06T15:33:40\",\"Id\":\"3f56f3e7-7098-45f2-b0fc-3cbad632120c\",\"Operation\":\"Add member to role.\",\"OrganizationId\":\"1c04d22a-c0e8-4bec-840b-f1a7fb855091\",\"RecordType\":8,\"ResultStatus\":\"Success\",\"UserKey\":\"10032001C8CDE20F@domain.com\",\"UserType\":0,\"Version\":1,\"Workload\":\"AzureActiveDirectory\",\"ObjectId\":\"testuser@domain.com\",\"UserId\":\"User1@domain.com\",\"AzureActiveDirectoryEventType\":1,\"ExtendedProperties\":[{\"Name\":\"additionalDetails\",\"Value\":\"{}\"},{\"Name\":\"extendedAuditEventCategory\",\"Value\":\"Role\"}],\"ModifiedProperties\":[{\"Name\":\"Role.ObjectID\",\"NewValue\":\"a6ce8341-0ca1-4f04-8127-7d891f95103c\",\"OldValue\":\"\"},{\"Name\":\"Role.DisplayName\",\"NewValue\":\"Global Administrator\",\"OldValue\":\"\"},{\"Name\":\"Role.TemplateId\",\"NewValue\":\"62e90394-69f5-4237-9190-012177145e10\",\"OldValue\":\"\"},{\"Name\":\"Role.WellKnownObjectName\",\"NewValue\":\"TenantAdmins\",\"OldValue\":\"\"}],\"Actor\":[{\"ID\":\"User1@domain.com\",\"Type\":5},{\"ID\":\"10032001C8CDE20F\",\"Type\":3},{\"ID\":\"User_892daa3c-4f3a-4a1a-b4ef-0fc987d61634\",\"Type\":2},{\"ID\":\"892daa3c-4f3a-4a1a-b4ef-0fc987d61634\",\"Type\":2},{\"ID\":\"User\",\"Type\":2}],\"ActorContextId\":\"1c04d22a-c0e8-4bec-840b-f1a7fb855091\",\"InterSystemsId\":\"b566a715-8e02-4c1d-9b69-86c18da3d194\",\"IntraSystemId\":\"182829f7-59da-47fd-95e2-466566f6ad47\",\"SupportTicketId\":\"\",\"Target\":[{\"ID\":\"User_255d2df7-4f0f-40e9-a14d-445c80b2cfbf\",\"Type\":2},{\"ID\":\"255d2df7-4f0f-40e9-a14d-445c80b2cfbf\",\"Type\":2},{\"ID\":\"User\",\"Type\":2},{\"ID\":\"testuser@domain.com\",\"Type\":5},{\"ID\":\"10032001C98098E7\",\"Type\":3}],\"TargetContextId\":\"1c04d22a-c0e8-4bec-840b-f1a7fb855091\"}",
      "code": "AzureActiveDirectory",
      "provider": "AzureActiveDirectory",
      "kind": "event",
      "action": "Add member to role.",
      "id": "3f56f3e7-7098-45f2-b0fc-3cbad632120c",
      "type": [
        "info"
      ],
      "category": [
        "web"
      ],
      "dataset": "o365.audit",
      "outcome": "success"
    },
    "user": {
      "domain": "domain.com",
      "name": "User1",
      "id": "User1@domain.com",
      "email": "User1@domain.com",
      "target": {
        "domain": "domain.com",
        "name": "testuser",
        "id": "testuser@domain.com",
        "email": "testuser@domain.com"
      }
    }
  },
  "fields": {
    "o365.audit.SupportTicketId": [
      ""
    ],
    "o365.audit.ModifiedProperties.Role_DisplayName.OldValue": [
      ""
    ],
    "elastic_agent.version": [
      "7.16.2"
    ],
    "event.category": [
      "web"
    ],
    "o365.audit.UserId": [
      "User1@domain.com"
    ],
    "o365.audit.ModifiedProperties.Role_ObjectID.OldValue": [
      ""
    ],
    "user.target.id": [
      "testuser@domain.com"
    ],
    "o365.audit.ExtendedProperties.additionalDetails": [
      "{}"
    ],
    "o365.audit.TargetContextId": [
      "1c04d22a-c0e8-4bec-840b-f1a7fb855091"
    ],
    "user.target.email": [
      "testuser@domain.com"
    ],
    "o365.audit.ModifiedProperties.Role_TemplateId.OldValue": [
      ""
    ],
    "agent.name": [
      "Shipper"
    ],
    "host.name": [
      "domain.com"
    ],
    "event.agent_id_status": [
      "verified"
    ],
    "event.kind": [
      "event"
    ],
    "o365.audit.Actor.Type": [
      5,
      3,
      2,
      2,
      2
    ],
    "event.outcome": [
      "success"
    ],
    "user.id": [
      "User1@domain.com"
    ],
    "cloud.region": [
      "eastus"
    ],
    "o365.audit.ModifiedProperties.Role_DisplayName.NewValue": [
      "Global Administrator"
    ],
    "input.type": [
      "o365audit"
    ],
    "agent.hostname": [
      "Shipper"
    ],
    "o365.audit.ObjectId": [
      "testuser@domain.com"
    ],
    "data_stream.type": [
      "logs"
    ],
    "o365.audit.ExtendedProperties.extendedAuditEventCategory": [
      "Role"
    ],
    "tags": [
      "forwarded",
      "o365-audit",
      "preserve_original_event"
    ],
    "user.target.name": [
      "testuser"
    ],
    "related.user": [
      "User1",
      "testuser"
    ],
    "event.provider": [
      "AzureActiveDirectory"
    ],
    "cloud.provider": [
      "azure"
    ],
    "cloud.machine.type": [
      "Standard_D2s_v3"
    ],
    "event.code": [
      "AzureActiveDirectory"
    ],
    "cloud.service.name": [
      "Virtual Machines"
    ],
    "agent.id": [
      "21d0ccfd-4784-4126-8445-cc4c3437e6e4"
    ],
    "o365.audit.AzureActiveDirectoryEventType": [
      "1"
    ],
    "ecs.version": [
      "1.12.0"
    ],
    "o365.audit.RecordType": [
      "8"
    ],
    "organization.id": [
      "1c04d22a-c0e8-4bec-840b-f1a7fb855091"
    ],
    "agent.version": [
      "7.16.2"
    ],
    "o365.audit.ActorContextId": [
      "1c04d22a-c0e8-4bec-840b-f1a7fb855091"
    ],
    "o365.audit.CreationTime": [
      "2022-01-06T15:33:40"
    ],
    "o365.audit.ModifiedProperties.Role_WellKnownObjectName.OldValue": [
      ""
    ],
    "user.name": [
      "User1"
    ],
    "o365.audit.UserKey": [
      "10032001C8CDE20F@domain.com"
    ],
    "o365.audit.Version": [
      "1"
    ],
    "cloud.instance.id": [
      "3dc43ef6-ee8b-4e06-895b-d134cb4513c8"
    ],
    "agent.type": [
      "filebeat"
    ],
    "event.module": [
      "o365"
    ],
    "user.email": [
      "User1@domain.com"
    ],
    "user.target.domain": [
      "domain.com"
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "o365.audit.InterSystemsId": [
      "b566a715-8e02-4c1d-9b69-86c18da3d194"
    ],
    "user.domain": [
      "domain.com"
    ],
    "host.id": [
      "1c04d22a-c0e8-4bec-840b-f1a7fb855091"
    ],
    "o365.audit.Target.Type": [
      2,
      2,
      2,
      5,
      3
    ],
    "elastic_agent.id": [
      "21d0ccfd-4784-4126-8445-cc4c3437e6e4"
    ],
    "o365.audit.IntraSystemId": [
      "182829f7-59da-47fd-95e2-466566f6ad47"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "o365.audit.Target.ID": [
      "User_255d2df7-4f0f-40e9-a14d-445c80b2cfbf",
      "255d2df7-4f0f-40e9-a14d-445c80b2cfbf",
      "User",
      "testuser@domain.com",
      "10032001C98098E7"
    ],
    "o365.audit.UserType": [
      "0"
    ],
    "o365.audit.ModifiedProperties.Role_TemplateId.NewValue": [
      "62e90394-69f5-4237-9190-012177145e10"
    ],
    "o365.audit.ModifiedProperties.Role_ObjectID.NewValue": [
      "a6ce8341-0ca1-4f04-8127-7d891f95103c"
    ],
    "o365.audit.ResultStatus": [
      "Success"
    ],
    "event.ingested": [
      "2022-01-06T15:59:15.000Z"
    ],
    "event.action": [
      "Add member to role."
    ],
    "@timestamp": [
      "2022-01-06T15:33:40.000Z"
    ],
    "o365.audit.ModifiedProperties.Role_WellKnownObjectName.NewValue": [
      "TenantAdmins"
    ],
    "event.type": [
      "info"
    ],
    "data_stream.dataset": [
      "o365.audit"
    ],
    "agent.ephemeral_id": [
      "684b66ad-a186-417e-b4a5-eb37263289d1"
    ],
    "event.id": [
      "3f56f3e7-7098-45f2-b0fc-3cbad632120c"
    ],
    "event.dataset": [
      "o365.audit"
    ],
    "o365.audit.Actor.ID": [
      "User1@domain.com",
      "10032001C8CDE20F",
      "User_892daa3c-4f3a-4a1a-b4ef-0fc987d61634",
      "892daa3c-4f3a-4a1a-b4ef-0fc987d61634",
      "User"
    ],
    "cloud.instance.name": [
      "Shipper"
    ]
  }
}
```

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

Resolves #1685 

## Summary

Adds two rules using different Integrations data sources (O365/Azure).

In Azure Active Directory (Azure AD), permissions to manage resources are assigned using Roles. The Global Administrator
is a role that enables users to have access to all administrative features in Azure AD and services that use Azure
Active Directory identities like the Microsoft 365 Defender portal, the Microsoft 365 compliance center, Exchange,
SharePoint Online, and Skype for Business Online. Attackers can add users as Global Administrators to maintain access
and manage all subscriptions and their settings and resources.

### CODE REVIEW

**Review by Samirbous — APPROVED** [review]

File: rules/integrations/azure/persistence_azure_global_administrator_role_assigned.toml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+rule_id = "04c5a96f-19c5-44fd-9571-a0b033f9086f"
+severity = "medium"
+tags = ["Elastic", "Cloud", "Azure", "Continuous Monitoring", "SecOps", "Identity and Access"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.dataset:azure.auditlogs and azure.auditlogs.properties.category:RoleManagement and
+azure.auditlogs.operation_name : "Add member to role" and
+azure.auditlogs.properties.target_resources.0.modified_properties.1.new_value: "\"Global Administrator\""
```
Comment (brokensound77) [inline rules/integrations/azure/persistence_azure_global_administrator_role_assigned.toml:37]: nit:

```suggestion
azure.auditlogs.operation_name:"Add member to role" and
azure.auditlogs.properties.target_resources.0.modified_properties.1.new_value:"\"Global Administrator\""
```

File: rules/integrations/o365/persistence_microsoft_365_global_administrator_role_assign.toml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+author = ["Elastic"]
+description = """
+In Azure Active Directory (Azure AD), permissions to manage resources are assigned using Roles. The Global Administrator
+is a role that enables users to have access to all administrative features in Azure AD and services that use Azure
+Active Directory identities like the Microsoft 365 Defender portal, the Microsoft 365 compliance center, Exchange,
+SharePoint Online, and Skype for Business Online. Attackers can add users as Global Administrators to maintain access
+and manage all subscriptions and their settings and resources.
+"""
+from = "now-25m"
+index = ["filebeat-*", "logs-azure*"]
```
Comment (brokensound77) [inline rules/integrations/o365/persistence_microsoft_365_global_administrator_role_assign.toml:17]:
```suggestion
index = ["filebeat-*", "logs-o365*"]
```

File: rules/integrations/o365/persistence_microsoft_365_global_administrator_role_assign.toml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+risk_score = 47
+rule_id = "88671231-6626-4e1b-abb7-6e361a171fbb"
+severity = "medium"
+tags = ["Elastic", "Cloud", "Microsoft 365", "Continuous Monitoring", "SecOps", "Identity and Access"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.dataset:o365.audit and event.code:"AzureActiveDirectory" and event.action: "Add member to role." and
+o365.audit.ModifiedProperties.Role_DisplayName.NewValue: "Global Administrator"
```
Comment (brokensound77) [inline rules/integrations/o365/persistence_microsoft_365_global_administrator_role_assign.toml:36]: nit:

```suggestion
event.dataset:o365.audit and event.code:"AzureActiveDirectory" and event.action:"Add member to role." and
o365.audit.ModifiedProperties.Role_DisplayName.NewValue:"Global Administrator"
```

Also this `NewValue` does not include quotes like the azure rule - is that intentional?

File: rules/integrations/o365/persistence_microsoft_365_global_administrator_role_assign.toml [hunk 4]
```diff
@@ -0,0 +1,56 @@
+[metadata]
+creation_date = "2022/01/06"
+integration = "o365"
+maturity = "production"
+updated_date = "2022/01/06"
+
+[rule]
+author = ["Elastic"]
+description = """
+In Azure Active Directory (Azure AD), permissions to manage resources are assigned using Roles. The Global Administrator
```
Comment (brokensound77) [inline rules/integrations/o365/persistence_microsoft_365_global_administrator_role_assign.toml:10]: Does this need to be updated to reflect 0365?

**Review by brokensound77 — APPROVED** [review]
A few suggestions and questions, but otherwise LGTM 👍

File: rules/integrations/o365/persistence_microsoft_365_global_administrator_role_assign.toml (hunk 3 above)
Comment (w0rk3r) [reply rules/integrations/o365/persistence_microsoft_365_global_administrator_role_assign.toml]: Yeah, the azure value contains quotes:
image
While this one don't

File: rules/integrations/o365/persistence_microsoft_365_global_administrator_role_assign.toml (hunk 4 above)
Comment (w0rk3r) [reply rules/integrations/o365/persistence_microsoft_365_global_administrator_role_assign.toml:10]: Both use Azure AD to manage these Roles

Comment (w0rk3r) [reply rules/integrations/o365/persistence_microsoft_365_global_administrator_role_assign.toml:10]: It is basically the same action, resulting in the same log, but coming from 2 different data sources

**Pull request merged**